### PR TITLE
Adding ClamAv scanner for scanning files for virus

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Adding ClamAv scanner for scanning files for virus

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by akroon3r** [comment]
@alexjoybc typo in your pom

Failed to execute goal on project efiling-api: Could not resolve dependencies for project ca.bc.gov.open.jag:efiling-api:jar:0.4.1-SNAPSHOT: Could not find artifact ca.bc.gov.open:spring-clamav-starter:jar:0.1.3 in central (https://repo.maven.apache.org/maven2) -> [Help 1]

Nexus has 0.1.4, not 0.4.1

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 1]
```diff
@@ -204,21 +206,18 @@ public SubmissionApiDelegateImpl(
                             submissionService.generateFromRequest(xTransactionId, submissionId, actualUserId.get(), generateUrlRequest),
                             navigationProperties.getBaseUrl()));
             logger.info("successfully generated return url.");
-        }
-        catch (CSOHasMultipleAccountException e)   {
+        } catch (CSOHasMultipleAccountException e) {
             logger.warn(e.getMessage(), e);
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:210]: out of scope: is this a warn? or is this an error.

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/UploadSubmissionDocumentsTest.java [hunk 2]
```diff
@@ -120,6 +156,26 @@ public void withIoExceptionShouldReturnInternalServer() {
         Assertions.assertEquals(DOCUMENT_STORAGE_FAILURE.getErrorMessage(), ((EfilingError)actual.getBody()).getMessage());
     }
 
+    @Test
+    @DisplayName("502: with ioException should return 502")
```
Comment (akroon3r) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/UploadSubmissionDocumentsTest.java:160]: does this need to be more descriptive? seems circular

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java [hunk 3]
```diff
@@ -75,24 +83,51 @@ public void setUp() throws IOException {
...
+
         ResponseEntity<UploadSubmissionDocumentsResponse> actual = sut.uploadAdditionalSubmissionDocuments(TestHelpers.CASE_1, UUID.randomUUID(), files);
 
         Assertions.assertEquals(HttpStatus.OK, actual.getStatusCode());
         Assertions.assertEquals(TestHelpers.CASE_1, actual.getBody().getSubmissionId());
         Assertions.assertEquals(new BigDecimal(2), actual.getBody().getReceived());
     }
 
+    @Test
+    @DisplayName("400: with non pdf files should return bad request")
```
Comment (akroon3r) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java:111]: isn't this already tested?

File: src/backend/efiling-api/src/test/resources/test.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+blorb
```
Comment (akroon3r) [inline src/backend/efiling-api/src/test/resources/test.txt:1]: yes. blorb. yes.

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java [hunk 5]
```diff
@@ -75,24 +83,51 @@ public void setUp() throws IOException {
...
+    public void withNonPdfFilesShouldReturnOk() throws IOException, VirusDetectedException {
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java:112]:
```suggestion
    public void withNonPdfFilesShouldReturnBadRequest() throws IOException, VirusDetectedException {
```

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java [hunk 6]
```diff
@@ -75,24 +83,51 @@ public void setUp() throws IOException {
 
         Mockito.when(submissionStoreMock.get(Mockito.eq(TestHelpers.CASE_1), Mockito.any())).thenReturn(Optional.of(submission));
 
-        sut = new SubmissionApiDelegateImpl(submissionServiceMock, accountServiceMock, generateUrlResponseMapperMock, navigationProperties, submissionStoreMock, documentStoreMock);
+        sut = new SubmissionApiDelegateImpl(submissionServiceMock, accountServiceMock, generateUrlResponseMapperMock, navigationProperties, submissionStoreMock, documentStoreMock, clamAvServiceMock);
     }
 
     @Test
     @DisplayName("200: with files should return ok")
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java:90]:
```suggestion
    @DisplayName("200: with pdf files should return ok")
```

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java [hunk 7]
```diff
@@ -75,24 +83,51 @@ public void setUp() throws IOException {
...
-    public void withFilesShouldReturnOk() {
+    public void withFilesShouldReturnOk() throws IOException, VirusDetectedException {
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/uploadAdditionalSubmissionDocumentsTest.java:91]:
```suggestion
    public void withPdfFilesShouldReturnOk() throws IOException, VirusDetectedException {
```

**Pull request merged**